synadm/api.py

```diff
@@ -28,12 +28,39 @@
 https://matrix.org/docs/spec/#matrix-apis.
 """
 
-import requests
-from http.client import HTTPConnection
 import datetime
 import json
-import urllib.parse
 import re
+from typing import Optional, Union, Dict, List, Any
+import urllib.parse
+from http.client import HTTPConnection
+import traceback
+
+import requests
+from requests.exceptions import ConnectionError
+
+
+def log_fatal_exit(error, logger, message=None):
+    """Log a fatal error and exit synadm.
+
+    Args:
+        error: A Python exception.
+        logger: A Python logger, with info and fatal methods
+        message: Message to use instead of "synadm exited due to a fatal
+            error."
+    """
+    if message is None:
+        message = "synadm exited due to a fatal error."
+
+    # split by new lines that's already included
```

Comment:
```suggestion
    # format_exception() returns a list of strings. join it into a single string again. 
```

Would this still make sense?

I would like this see how this looks like in action. Can you give me a hint on how to produce a traceback. I can't seem to find a way to trigger one. Thanks!